### PR TITLE
context: use autogenerated name to improve multi-tenancy

### PR DESCRIPTION
**Pull request by bloom1**
Depends-On: https://github.com/wazo-platform/xivo-manage-db/pull/210
Depends-On: https://github.com/wazo-platform/xivo-dao/pull/233

### CODE REVIEW

**Comment by bloom1** [comment]
recheck

**Comment by bloom1** [comment]
recheck

**Comment by bloom1** [comment]
recheck

**Comment by bloom1** [comment]
recheck

**Comment by bloom1** [comment]
recheck

**Comment by bloom1** [comment]
recheck

**Comment by bloom1** [comment]
recheck

**Comment by bloom1** [comment]
recheck

**Review by fblackburn1 — COMMENTED** [review]
(didn't have to complete review, but I don't want to lose my comment, so I submit this partial review)

**Comment by bloom1** [comment]
recheck

**Review by fblackburn1 — COMMENTED** [review]
Other code LGTM
small tear about using a lot of `aux()` function instead of using context manager:
 `with fixtures.extension(context=...)`
But I can understand that context manager on multi lines is ugly to read.
Maybe using ExitStack (ex: [test_extensions.py](https://github.com/wazo-platform/wazo-confd/blob/master/integration_tests/suite/base/test_extensions.py#L396)
Or waiting on python3.10 to [parentheses](https://docs.python.org/3/whatsnew/3.10.html#parenthesized-context-managers)
Anyways, just wanted to share with you. It's ok for me

File: wazo_confd/plugins/context/schema.py [hunk 1]
```diff
@@ -92,6 +79,18 @@ def create_objects(self, data, **kwargs):
                 data[key] = [ContextNumbers(**d) for d in data[key]]
         return data
 
+    # DEPRECATED 23.10
```
Comment (fblackburn1) [inline wazo_confd/plugins/context/schema.py:82]:
```suggestion
    # DEPRECATED 23.11
```

**Review by fblackburn1 — CHANGES_REQUESTED** [review]

**Review by fblackburn1 — CHANGES_REQUESTED** [review]
We still have some context limited to 39
```
helpers/destination.py:CONTEXT_REGEX = r'^[a-zA-Z0-9_-]{1,39}$'
```

**Comment by bloom1** [comment]
recheck

**Comment by bloom1** [comment]
recheck

**Review by fblackburn1 — APPROVED** [review]

File: wazo_confd/plugins/context/resource.py [hunk 2]
```diff
@@ -14,13 +15,29 @@
 class ContextList(ListResource):
     model = Context
     schema = ContextSchema
+    context_name_fmt = 'ctx-{tenant_slug}-{context_uuid}'
```
Comment (pc-m) [inline wazo_confd/plugins/context/resource.py:18]: Could you add the context type or an abbreviation for the type after the slug?
`ctx-{tenant_slug}-{context_type}-{context-uuid}'

The result would look like
`ctx-customer1-internal-1d07eb9a-0389-46da-9800-2d95dbce1e7e`
`ctx-customer1-incall-1d07eb9a-0389-46da-9800-2d95dbce1e7d`
`ctx-customer1-outcall-1d07eb9a-0389-46da-9800-2d95dbce1e7c`

Comment (fblackburn1) [reply wazo_confd/plugins/context/resource.py]: fixed

**Review by fblackburn1 — DISMISSED** [review]

**Review by fblackburn1 — APPROVED** [review]

**Review by fblackburn1 — APPROVED** [review]

**Pull request merged**